src/Cli/dotnet/commands/dotnet-sln/remove/Program.cs

```diff
@@ -77,7 +76,18 @@ private async Task RemoveProjectsAsync(string solutionFileFullPath, IEnumerable<
             foreach (var projectPath in projectPaths)
             {
                 var project = solution.FindProject(projectPath);
-                if (project != null)
+                // If the project is not found, try to find it by name without extension
+                if (project is null && !Path.HasExtension(projectPath))
+                {
+                    try 
+                    {
+                        project = solution.SolutionProjects.SingleOrDefault(p => Path.GetFileNameWithoutExtension(p.DisplayName) == projectPath);
```

Comment:
```suggestion
                        project = solution.SolutionProjects.SingleOrDefault(p => Path.GetFileNameWithoutExtension(p.DisplayName).Equals(projectPath));
```

Better to use .Equals for string comparisons. This also means you can (and should) add StringComparison.OrdinalIgnoreCase on case-insensitive OSs (broadly, windows)